plugins/modules/monitoring/alerta_customer.py

```diff
@@ -0,0 +1,194 @@
+      - The Alerta API endpoint.
+    required: true
+    type: str
+  api_username:
+    description:
+      - The username for the API using basic auth.
+    type: str
+  api_password:
+    description:
+      - The password for the API using basic auth.
+    type: str
+  api_key:
+    description:
+      - The access token for the API.
+    type: str
+  state:
+    description:
+      - Whether the customer should exist or not.
+    type: str
+    choices: [ absent, present ]
+    default: present
+'''
+
+EXAMPLES = """
+- name: Create customer
+  community.general.alerta_customer:
+    alerta_url: https://alerta.example.com
+    api_username: admin@example.com
+    api_password: password
+    customer: Developer
+    match: dev@example.com
+
+- name: Delete customer
+  community.general.alerta_customer:
+    alerta_url: https://alerta.example.com
+    api_username: admin@example.com
+    api_password: password
+    customer: Developer
+    match: dev@example.com
+    state: absent
+"""
+
+RETURN = """
+msg:
+  description:
+    - Success or failure message.
+  returned: always
+  type: str
+  sample: Customer customer1 created
+response:
+  description:
+    - The response from the API.
+  returned: always
+  type: dict
+"""
+
+from ansible.module_utils.urls import fetch_url, basic_auth_header
+from ansible.module_utils.basic import AnsibleModule
+
+
+class AlertaInterface(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.state = module.params['state']
+        self.customer = module.params['customer']
+        self.match = module.params['match']
+        self.alerta_url = module.params['alerta_url']
+        self.headers = {"Content-Type": "application/json"}
+
+        if module.params.get('api_key', None):
+            self.headers["Authorization"] = "Key %s" % module.params['api_key']
+        else:
+            self.headers["Authorization"] = basic_auth_header(module.params['api_username'], module.params['api_password'])
+
+    def send_request(self, url, data=None, method="GET"):
+        response, info = fetch_url(self.module, url, data=data, headers=self.headers, method=method)
+
+        status_code = info["status"]
+        if status_code == 401:
+            self.module.fail_json(failed=True, response=info, msg="Unauthorized to request '%s' on '%s'" % (method, url))
+        elif status_code == 403:
+            self.module.fail_json(failed=True, response=info, msg="Permission Denied for '%s' on '%s'" % (method, url))
+        elif status_code == 404:
+            self.module.fail_json(failed=True, response=info, msg="Not found for request '%s' on '%s'" % (method, url))
+        elif status_code == 200 or status_code == 201:
+            return self.module.from_json(response.read())
+        self.module.fail_json(failed=True, response=info, msg="Alerta API error with HTTP %d for %s" % (status_code, url))
+
+    def get_customers(self):
+        url = "%s/api/customers" % (self.alerta_url)
```

Comment:
I think it's better to make this a proper tuple:
```suggestion
        url = "%s/api/customers" % (self.alerta_url, )
```
(Same below.)

Reply:
removed tuple